stix2/base.py

```diff
@@ -76,6 +79,11 @@ def get_required_properties(properties):
 class _STIXBase(Mapping):
     """Base class for STIX object types"""
 
+    def get_class_version(self):
+        module_name = self.__class__.__module__
+        module_parts = module_name.split(".")
+        return module_parts[1]
+
```

Comment:
Yuuuuck.  An ugly dependency between STIX version and what module a class is in.  I had a similar need in `new_version`, where I needed to know which STIX version the object was so I could use the correct versioning precision rules.  But in that case, the object was fully formed and I could use the `spec_version` property.  Here, it seems you're not so lucky.

Could we just make all 2.1 objects extend a special 2.1 class and use isinstance()?  And the same for 2.0 objects?

Reply:
@clenk , @chisholm - I like this idea.  What I did was the best I could come up with after discussing it with Chris.  Let's make it independent of this PR, and I can use it once it is implemented and tested

Reply:
Agreed; we should use a 2.1 class with `isinstance()`, and fix it in a subsequent issue/PR.